File: fix_malformed_imports.py
#!/usr/bin/env python3
"""
Script to fix malformed import statements where Helmet import was inserted incorrectly.
"""

import os
import re
from pathlib import Path

def fix_malformed_imports_in_file(file_path):
    """Fix malformed import statements in a single file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has malformed imports
        if 'import {\nimport { Helmet }' in content:
            print(f"Fixing malformed imports in: {file_path}")
            
            # Fix the malformed import pattern
            content = re.sub(
                r'import \{\nimport \{ Helmet \} from \'react-helmet-async\';\n',
                r'import { Helmet } from \'react-helmet-async\';\nimport {\n',
                content
            )
            
            # Write the fixed content back
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix malformed imports."""
    workspace_root = Path('/workspace')
    app_dir = workspace_root / 'app'
    
    if not app_dir.exists():
        print("App directory not found!")
        return
    
    # Find all TSX files
    tsx_files = list(app_dir.rglob('*.tsx'))
    
    print(f"Found {len(tsx_files)} TSX files to check")
    
    fixed_count = 0
    for file_path in tsx_files:
        if fix_malformed_imports_in_file(file_path):
            fixed_count += 1
    
    print(f"Fixed malformed imports in {fixed_count} files")

if __name__ == "__main__":
    main()

File: fix-malformed-imports.js
import fs from 'fs';
import path from 'path';

// Get all TypeScript/JavaScript files
const getAllFiles = (dir, extensions = ['.ts', '.tsx', '.js', '.jsx']) => {
  let files = [];
  const items = fs.readdirSync(dir);
  
  for (const item of items) {
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    
    if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
      files = files.concat(getAllFiles(fullPath, extensions));
    } else if (extensions.some(ext => item.endsWith(ext))) {
      files.push(fullPath);
    }
  }
  
  return files;
};

// Fix malformed imports in a file
const fixMalformedImports = (filePath) => {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Fix malformed import statements that have code on the same line
    const malformedImportRegex = /import\s+{[^}]+}\s+from\s+['"][^'"]+['"];\s*[^/]/g;
    
    content = content.replace(malformedImportRegex, (match) => {
      // Extract the import part and the code part
      const importEnd = match.indexOf(';');
      const importStatement = match.substring(0, importEnd + 1);
      const remainingCode = match.substring(importEnd + 1).trim();
      
      if (remainingCode) {
        modified = true;
        return importStatement + '\n' + remainingCode;
      }
      return match;
    });
    
    // Fix imports that are missing semicolons and have code after
    const missingSemicolonRegex = /import\s+{[^}]+}\s+from\s+['"][^'"]+['"]\s+[^/]/g;
    
    content = content.replace(missingSemicolonRegex, (match) => {
      const importEnd = match.indexOf("'");
      const lastQuote = match.lastIndexOf("'");
      if (lastQuote > importEnd) {
        const importStatement = match.substring(0, lastQuote + 1) + ';';
        const remainingCode = match.substring(lastQuote + 1).trim();
        
        if (remainingCode) {
          modified = true;
          return importStatement + '\n' + remainingCode;
        }
      }
      return match;
    });
    
    // Fix imports that are concatenated with other code
    const concatenatedImportRegex = /import\s+{[^}]+}\s+from\s+['"][^'"]+['"];\s*const\s+/g;
    
    content = content.replace(concatenatedImportRegex, (match) => {
      modified = true;
      return match.replace('; const', ';\nconst');
    });
    
    // Clean up multiple empty lines
    content = content.replace(/\n\s*\n\s*\n/g, '\n\n');
    
    if (modified) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed malformed imports in: ${filePath}`);
    }
    
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
  }
};

// Main execution
console.log('Fixing malformed imports...');

const files = getAllFiles('./app');
let fixedCount = 0;

files.forEach(file => {
  try {
    fixMalformedImports(file);
    fixedCount++;
  } catch (error) {
    console.error(`Error with file ${file}:`, error.message);
  }
});

console.log(`Processed ${fixedCount} files`);

// Also fix the main App.tsx
fixMalformedImports('./App.tsx');

console.log('Malformed imports fix completed!');